main.go

```diff
@@ -61,6 +123,53 @@ func main() {
 	http.HandleFunc("/thermostats", p.ServeThermostatsList)
 	http.HandleFunc("/thermostat", p.ServeThermostat)
 
-	log.Printf("Starting on %v", *addr)
-	log.Fatal(http.ListenAndServe(*addr, nil))
+	log.Printf("Starting on %v", hostPort)
+	return http.ListenAndServe(hostPort, nil)
+}
+
+func doRegister(c *cli.Context) error {
+	storePath := c.String("store")
+	if storePath == "" {
+		cli.ShowCommandHelpAndExit(c, c.Command.Name, 1)
+	}
+	apiKey := c.String("api_key")
+	if apiKey == "" {
+		cli.ShowCommandHelpAndExit(c, c.Command.Name, 1)
+	}
+
+	resp, err := http.Get(fmt.Sprintf(authURLTemplate, apiKey))
+	if err != nil {
+		return cli.Exit(fmt.Errorf("failed initializing Pin Authentication: %v", err), 1)
```

Comment:
nit: make error capitalization consistent. sometimes you start with `failed`, other times with `Failed`